console/src/main/java/com/alibaba/nacos/console/controller/UserController.java

```diff
@@ -146,6 +155,23 @@ public Object updateUser(@RequestParam String username, @RequestParam String new
         
         return new RestResult<>(200, "update user ok!");
     }
+
+    private boolean hasPermission(String username, HttpServletRequest request) {
+        if (!authConfigs.isAuthEnabled()) {
+            return true;
+        }
+        if (request.getAttribute(RequestUtil.NACOS_USER_KEY) == null) {
+            return false;
+        }
+
+        NacosUser user = (NacosUser) request.getAttribute(RequestUtil.NACOS_USER_KEY);
+        // admin
+        if (user.isGlobalAdmin()) {
+            return true;
+        }
```

Comment:
I think this implementation will cause security problem. If you get admin from user input, users can mock or inject this value so that they can change others' password.

Reply:
The current version also has this problem, because admin can modify the password of other users.I just kept this feature.

![image](https://user-images.githubusercontent.com/43994656/102998193-abb76800-4561-11eb-8067-c73c533ee1ab.png)

Reply:
But current judge admin has auth to change other password is used the role queried by datasource.

Your implementation is used the request input. If I'm not admin, but I call this API with myself input. I will change other's password.

Reply:
抱歉，我不是很明白您“I call this API with myself input. I will change other's password”这里表达的意思，因为我看代码在`com.alibaba.nacos.core.auth.AuthFilter#doFilter`这里会调用方法`com.alibaba.nacos.console.senicurity.nacos.NacosAuthManager#login`,这里它将设置NACOS_USER_KEY
```
        List<RoleInfo> roleInfoList = roleService.getRoles(username);
        if (roleInfoList != null) {
            for (RoleInfo roleInfo : roleInfoList) {
                if (roleInfo.getRole().equals(NacosRoleServiceImpl.GLOBAL_ADMIN_ROLE)) {
                    user.setGlobalAdmin(true);
                    break;
                }
            }
        }
        req.setAttribute(RequestUtil.NACOS_USER_KEY, user);
```
在我的理解里，我在后续从request中取出的NACOS_USER_KEY是已经认证过后的用户信息。或许有哪些地方我没有考虑到的，会造成您所说的安全问题。
如是确实存在这个问题，我在这里认证的时候调用`com.alibaba.nacos.console.senicurity.nacos.NacosAuthManager#login`这个方法获取用户信息，这样应该保险的。

Reply:
如果是从AuthFilter中覆盖的应该也可以。

Reply:
那现在我需要做什么呢 : )

Reply:
上面那个问题， 需不需要换成工具类判断？

Reply:
done